packages/dds/tree/src/tree/anchorSet.ts

```diff
@@ -240,6 +244,53 @@ export class AnchorSet {
             dstPath.removeRef();
         }
     }
+
+    /**
+     * Updates the anchors according to the changes described in the given delta
+     */
+    public applyDelta(delta: Delta.Root): void {
+        let parentField: FieldKey | undefined;
+        let parent: UpPath | undefined;
+        const moveTable = new Map<Delta.MoveId, UpPath>();
+
+        const visitor = {
+            onDelete: (start: number, count: number): void => {
+                assert(parentField !== undefined, "Must be in a field to delete");
+                this.moveChildren(count, { parent, parentField, parentIndex: start }, undefined);
+            },
+            onInsert: (start: number, content: Delta.ProtoNode[]): void => {
+                assert(parentField !== undefined, "Must be in a field to insert");
+                this.moveChildren(content.length, undefined, { parent, parentField, parentIndex: start });
+            },
+            onMoveOut: (start: number, count: number, id: Delta.MoveId): void => {
+                assert(parentField !== undefined, "Must be in a field to move out");
+                moveTable.set(id, { parent, parentField, parentIndex: start });
+            },
+            onMoveIn: (start: number, count: number, id: Delta.MoveId): void => {
+                assert(parentField !== undefined, "Must be in a field to move in");
+                this.moveChildren(count, moveTable.get(id), { parent, parentField, parentIndex: start });
```

Comment:
Defend against a bug upstream being silently interpreted as an insert.
```suggestion
                const srcPath = moveTable.get(id) ?? fail("Must visit a move in after its move out");
                this.moveChildren(count, srcPath, { parent, parentField, parentIndex: start });
```